### PR TITLE
Typed supervision did not handle stopped behaviors correctly #25156

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Fixes #25156

**Comment by chbatey** [comment]
PLS BUILD

**Review by chbatey — APPROVED** [review]
LGTM, validation failed for ssh issue

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala [hunk 1]
```diff
@@ -690,4 +691,61 @@ class SupervisionSpec extends ActorTestKit with TypedAkkaSpecWithShutdown {
...
+        val actor = spawn(
+          Behaviors.supervise[Command](
+            Behaviors.setup(_ ⇒
+              Behaviors.stopped)
+          ).onFailure(strategy)
+        )
+        TestProbe().expectTerminated(actor, 3.second)
+      }
+
+      // this test doesn't make sense for Resume since there will be no second setup
```
Comment (johanandren) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala]: oh, right!

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
I count my review here.

**Pull request merged**